awscli/clidocs.py

```diff
@@ -179,6 +187,11 @@ def doc_option(self, arg_name, help_command, **kwargs):
         doc.style.dedent()
         doc.style.new_paragraph()
 
+    def doc_global_option(self, help_command, **kwargs):
```

Comment:
One thing that I was noticing was that for some of the custom commands where it was intermediary command to another was displaying the global arguments. For example, the CLI was waiter commands where you can specify `aws ec2 wait <name-of-waiter>`. These intermediary commands are more for providing organization/hierarchy and cannot be successfully invoked directly (e.g. `aws ec2 wait`) and so it does not make sense to include the global options as part of the documentation (e.g. running `aws ec2 wait help`) similar to what we do for service commands.

To address this, we should probably override the `doc_global_option` for the base custom command doc handler: https://github.com/aws/aws-cli/blob/f39908a51c2843e146c56b2c2625b63f15312ad0/awscli/customizations/commands.py#L363  In overriding it, we should have the logic check to see if there is anything in the command table and only document global arguments if there are no commands present.